examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1520 @@
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "           rule  rule_id         label  label_id\n",
+       "0      atelecta        0   atelectasis         0\n",
+       "1      collapse        1   atelectasis         0\n",
+       "2  cardiomegaly        2  cardiomegaly         1\n",
+       "3     the heart        3  cardiomegaly         1\n",
+       "4    heart size        4  cardiomegaly         1"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#building the dataframe \"rules\"\n",
+    "rules = pd.DataFrame([i for i in itertools.chain.from_iterable(mentions['rule'])], columns = [\"rule\"])\n",
+    "rules['rule_id'] = range(len(rules))\n",
+    "rules['label'] = np.concatenate([np.repeat(mentions['label'][i], len(mentions['rule'][i])) for i in range(14)])\n",
+    "rules['label_id'] = [labels2ids[rules['label'][i]] for i in range(len(rules))]\n",
+    "rules.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "T matrix: [[1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]]\n",
+      "shape: (148, 14)\n"
+     ]
+    }
+   ],
+   "source": [
+    "rule2rule_id = dict(zip(rules[\"rule\"], rules[\"rule_id\"]))\n",
+    "rule2label = dict(zip(rules[\"rule_id\"], rules[\"label_id\"]))\n",
+    "\n",
+    "def get_mapping_rules_labels_t(rule2label: Dict, num_classes: int) -> np.ndarray:\n",
+    "    \"\"\" Function calculates t matrix (rules x labels) using the known correspondence of relations to decision rules \"\"\"\n",
+    "    mapping_rules_labels_t = np.zeros([len(rule2label), num_classes])\n",
+    "    for rule, labels in rule2label.items():\n",
+    "        mapping_rules_labels_t[rule, labels] = 1\n",
+    "    return mapping_rules_labels_t\n",
+    "\n",
+    "mapping_rules_labels_t = get_mapping_rules_labels_t(rule2label, len(labels2ids))\n",
+    "print(\"T matrix:\", mapping_rules_labels_t[0:5,:])\n",
+    "print(\"shape:\", mapping_rules_labels_t.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We derive our T matrix with 148 rules for 14 classes. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we want to check if there are any rules which assign to the same class. Indeed, there is one case: rule \"defib\" builds a rule for two different classes. In our T matrix, \"defib\" occures in two rows, since knodle cannot handle one rule assigned to multiple classes. "
```

Comment:
Let's have it for now not as separate T rows, but as one rule related to two classes at the same time. It would work in the next Knodle release.

Reply:
Ok, then I need to delete one row and change one 0 to 1 in the according column in the remaining row manually.